### PR TITLE
Makes loadout gear tweaks a bit more modular

### PR DESCRIPTION
**Pull request by Alberyk**
This pr creates three flags that can be applied to any loadout entry:
- GEAR_HAS_COLOR_SELECTION: allows color selection of the item.
- GEAR_HAS_NAME_SELECTION: allows the item's name to be changed.
- GEAR_HAS_DESC_SELECTION: allows the item's description to be changed.

TODO:

- [x] See what items should be restricted from changing name description

### CODE REVIEW

File: html/changelogs/alberyk-namestuff.yml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+author: ChangeMe
```
Comment (Doxxmedearly) [inline html/changelogs/alberyk-namestuff.yml:1]: New alias?

File: code/modules/client/preference_setup/loadout/loadout_xeno/unathi.dm [hunk 2]
```diff
@@ -57,6 +54,7 @@
 	sort_category = "Xenowear - Unathi"
 	cost = 3
 	allowed_roles = list("Station Engineer", "Chief Engineer", "Atmospheric Technician", "Engineering Apprentice", "Roboticist")
+	flags = 0
```
Comment (Arrow768) [inline code/modules/client/preference_setup/loadout/loadout_xeno/unathi.dm:57]: Might be worth to create a `GEAR_NO_SELECTION` define (or something like that)

**Comment by Alberyk** [comment]
As I pointed out; validation is already handled by the New proc in the loadout datums itself.

**Comment by Arrow768** [comment]
Actually, hold on

**Pull request merged**